### PR TITLE
Some modifications that make the project can be built successfully

### PR DESCRIPTION
**Pull request by KB5201314**
see https://github.com/openSUSE/mentoring/issues/162#issuecomment-803853256

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: Rust
+
+on:
+  push:
+    branches: [ imlk-dev ]
```
Comment (oxr463) [inline .github/workflows/rust.yml:5]: Can these be updated to master?

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Setup components
+      run: rustup component add clippy rustfmt
+    - name: Check
+      shell: bash
+      run: |
```
Comment (oxr463) [inline .github/workflows/rust.yml:26]: Why not run the githook instead of duplicating the script here?

**Review by oxr463 — CHANGES_REQUESTED** [review]
Please see previous comments.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (KB5201314) [reply .github/workflows/rust.yml]: This is a mistake and has been fixed now

File: .github/workflows/rust.yml (hunk 2 above)
Comment (KB5201314) [reply .github/workflows/rust.yml:22]: The two scripts are slightly different. The bash script in `pre-commit` will only check staged files and format the files, while the script in `rust.yml` will check the files changed between `github.event.before` and `github.event.after` but does not format.

Or should I merge the two into one script file and put it in `scripts/check-format.sh`?

Comment (oxr463) [reply .github/workflows/rust.yml:22]: They can stay separate, but I like the idea of having a standalone script that can be run locally.

Comment (KB5201314) [reply .github/workflows/rust.yml:22]: > They can stay separate, but I like the idea of having a standalone script that can be run locally.

In order to run locally, how about running `cargo fmt` directly?

Comment (oxr463) [reply .github/workflows/rust.yml:22]: That works for me. But how would that be different from the pre-commit hook?

Comment (KB5201314) [reply .github/workflows/rust.yml:22]: > That works for me. But how would that be different from the pre-commit hook?

When submitting, the pre-commit hook script will automatically format the staged code and remind you in a friendly way. 
here is an example:

[asciicast](https://asciinema.org/a/iD8QrzTpTuoinFsq4Q462l9zx)

Comment (oxr463) [reply .github/workflows/rust.yml:22]: That looks good to me.

Comment (KB5201314) [reply .github/workflows/rust.yml:22]: > That looks good to me. Is this ready to merge?

Yes

**Comment by oxr463** [comment]
Closes: https://github.com/proot-me/proot-rs/issues/5

**Pull request merged**